jax/_src/compiler.py

```diff
@@ -53,6 +53,15 @@
     ),
 )
 
+_DISABLE_COMPILER_REMAT_OPTIMIZATION_PASS = config.bool_flag(
+    "jax_compiler_enable_remat_pass",
+    config.bool_env('JAX_COMPILER_ENABLE_REMAT_PASS', True),
+    help=(
+      'Disable the rematerialization HLO pass by default '
+      'Avoids having to pass --xla_disable_hlo_passes=rematerialization. '
```

Comment:
I personally prefer when the description is useful for users that have no knowledge at all of that.
What about something like:
```
Allow to enable the automatic rematerialization pass. This avoid to pass `--xla_disable_hlo_passes=rematerialization` to XLA. If you have OOM, this will automatically try to trade-off memory usage with extra computation. You can frequently get better results by doing it manually with jax.checkpoint.
```

Don't hesitate to re-write, I just wanted to give an example of what can be added, so new user don't need to ask any questions.

Reply:
Does JAX has a documentation about that talk about OOM? If so, maybe add the config there?

Reply:
Do you mean a config option related to OOMs?

Reply:
Maybe add the new option in that page: https://jax.readthedocs.io/en/latest/gpu_memory_allocation.html
This way, more users will find it?